typescript/ts-converter/src/ast/AstExpressionConverter.ts

```diff
@@ -324,6 +346,36 @@ export class AstExpressionConverter {
         }
     }
 
+    convertTemplateExpression(expression: ts.TemplateExpression): Expression {
+        let tokens: Array<TemplateTokenDeclaration> = [];
+
+        let head = expression.head.text;
+
+        tokens.push(this.createStringTemplateToken(head));
+        for (let span of expression.templateSpans) {
+            if (ts.isTemplateMiddle(span.literal)) {
+                let text = span.literal.text;
+
+                tokens.push(this.createExpressionTemplateToken(
+                    this.convertExpression(span.expression)
+                ));
+                tokens.push(this.createStringTemplateToken(text));
+            } else if (ts.isTemplateTail(span.literal)) {
+                let text = span.literal.text;
+
+                tokens.push(this.createExpressionTemplateToken(
+                    this.convertExpression(span.expression)
+                ));
+                tokens.push(this.createStringTemplateToken(text));
+            }
+        }
+        return this.createTemplateExpression(tokens)
+    }
+
+    convertNoSubstitutionTemplateLiteral(literal: ts.NoSubstitutionTemplateLiteral): Expression {
+        return this.createStringLiteralExpression(literal.getText()
+            .split('`').join( '"'))
```

Comment:
Here is example of problems with `.getText()`. Because token is returned with quote marks surrounded, in case it has non-standard quote marks, we will need to replace them with standard